ex02.exs

```diff
@@ -1,6 +1,32 @@
 
 defmodule Ex02 do
+  def new_counter(value \\ 0) do
+      {:ok, counter} = Agent.start_link(Counter, :new, [value])
+      counter
+  end
+
+  def next_value(counter) do
+    Agent.get_and_update(counter, Counter, :next_value, [])
+  end
+
+  @me TheCounter
+  def new_global_counter(value \\ 0) do
+    {:ok, counter} = Agent.start_link(Counter, :new, [value], name: @me)
+    counter
+  end
+
+  def global_next_value() do
+    Agent.get_and_update(@me, &Counter.next_value/1)
+  end
+end
 
+defmodule Counter do
```

Comment:
## +2

I'm giving a bonus here because it's a cute idea.

However, I'm not sure it's a good one, but the reason is mostly just opinion.

It seems to me that because `next_value/1` returns a `{state, state + 1}` tuple, this module assumes that it is being called in the context of `Agent.get_and_update`. This means that the module actually introduces coupling, rather than reducing it.